app/models/user.rb

```diff
@@ -1,9 +1,12 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable, :confirmable
```

Comment:
- [ ] Great work setting up all the modules for the devise gem. However, the essence of setting authentication in your application is to ensure that users authenticate who they are before having access to the application. This adds a layer of security and ensures that the user logging into the application is the actual person. When I run your application, I get direct access to the `Users#Index` page which shouldn't be. I had to manually input the URL for the sign-up page before creating a user. This should be done immediately after a user is directed to your root route. Kindly implement this and ensure that users don't have direct access to the application before authentication is done. 👍 🖱️ 

#### Hint: You can use the `  before_action :authenticate_user!` in the Users controller.

Reply:
Thank you very much.
I will implement it now

Reply:
Thanks again for the suggestion
I have updated it as suggested

![Screenshot 2023-11-09 002238](https://github.com/SnakyMz/Blog-Ruby/assets/115364362/86420301-0d8a-469f-badf-8bf2407ccce6)

Reply:
> Thanks again for the suggestion I have updated it as suggested
> 
> ![Screenshot 2023-11-09 002238](https://user-images.githubusercontent.com/115364362/281512293-86420301-0d8a-469f-badf-8bf2407ccce6.png)

You are welcome @SnakyMz